src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -66,7 +75,30 @@ const CriteriaView = ({ criteria, deleteCriteria }: CriteriaViewProps) => {
               case 'domain':
                 return h(Fragment, [strong([`${criteria.domainOption.category}:`]), ` ${criteria.name}`]);
               case 'list':
-                return h(Fragment, [strong([`${criteria.name}:`]), ` ${criteria.value.name}`]);
+                return h(Fragment, [
+                  strong([`${criteria.name}: ${_.flow(_.map('name'), _.join(','))(criteria.valuesSelected)}`]),
+                  h(Select, {
```

Comment:
It looks like this could parameterize on `Option`
```suggestion
                  h(Select<Option>, {
```

Reply:
The types appear to be super messed up for Select around multiselect inputs. I am going to file a tech debt ticket in our backlog for fixing that, since it is fairly wide-spanning.